.github/test-spec.yml

```diff
@@ -151,40 +151,54 @@
   - "modules/mbedtls/**/*"
 
 "CI-fem-test":
+  - "boards/nordic/**/*"
+  - "drivers/bluetooth/hci/**/*"
+  - "drivers/entropy/**/*"
+  - "dts/bindings/**/*"
+  - "include/zephyr/net/**/*"
+  - "include/zephyr/arch/**/*"
+  - "lib/libc/**/*"
+  - "lib/open-amp/**/*"
+  - "modules/hal_nordic/**/*"
+  - "modules/mbedtls/**/*"
+  - "modules/openthread/**/*"
+  - "modules/trusted-firmware-m/**/*"
   - "samples/net/sockets/echo_*/**/*"
-  - any:
-      - "!.github/**/*"
-      - "!doc/**/*"
-      - "!CODEOWNERS"
-      - "!LICENSE"
-      - "!**/*.rst"
-      - "!VERSION"
-      - "!SDK_VERSION"
-      - "!MAINTAINERS.yml"
-      - "!version.h.in"
-      - "!Jenkinsfile"
-      - "!**/*.md"
-      - "!tests/**/*"
-      - "!samples/**/*"
-      - "!scripts/**/*"
+  - "share/**/*"
+  - "soc/nordic/**/*"
+  - "subsys/net/**/*"
+  - "subsys/settings/**/*"
+  - "subsys/bluetooth/shell/**/*"
+  - "subsys/ipc/**/*"
+  - "Kconfig"
+  - "CMakeLists.txt"
+  - "!**/*.rst"
+  - "!**/*.md"
```

Comment:
With such pattern an update to any file not matching *.rst or *.md would trigger the build. I believe this is not a goal.
Perhaps you should just remove these negations.